man/man7/git-secret.7.ronn

```diff
@@ -5,88 +5,151 @@ git-secret - bash tool to store private data inside a git repo.
+1. [Get their `gpg` public-key](#using-gpg). **You won't need their secret key.**
 
 2. Import this key into your `gpg` setup (in ~/.gnupg or similar) by running `gpg --import KEY_NAME.txt`
 
-3. Now add this person to your secrets repo by running `git secret tell persons@email.id` 
+3. Now add this person to your secrets repo by running `git secret tell persons@email.id`
 (this will be the email address associated with the public key)
 
-4. The newly added user cannot yet read the encrypted files. Now, re-encrypt the files using 
-`git secret reveal; git secret hide -d`, and then commit and push the newly encrypted files. 
-(The -d options deletes the unencrypted file after re-encrypting it). 
+4. The newly added user cannot yet read the encrypted files. Now, re-encrypt the files using
+`git secret reveal; git secret hide -d`, and then commit and push the newly encrypted files.
+(The -d options deletes the unencrypted file after re-encrypting it).
 Now the newly added user be able to decrypt the files in the repo using `git-secret`.
 
-Note that it is possible to add yourself to the git-secret repo without decrypting existing files. 
-It will be possible to decrypt them after re-encrypting them with the new keyring. So, if you don't 
+Note that it is possible to add yourself to the git-secret repo without decrypting existing files.
+It will be possible to decrypt them after re-encrypting them with the new keyring. So, if you don't
 want unexpected keys added, you can configure some server-side security policy with the `pre-receive` hook.
 
+### Using gpg
+
+You can follow a quick gpg tutorial at https://www.devdungeon.com/content/gpg-tutorial. Here are the most useful commands to get started:
+
+To generate a RSA key-pair, run:
+
+```shell
+gpg --gen-key
+```
+
+To export your public key, run:
+
+```shell
+gpg --export your.email@address.com --armor > public-key.gpg
+```
+
+To import the public key of someone else (to share the secret with them for instance), run:
+
+```shell
+gpg --import public-key.gpg
+```
+
+Be sure to use a secure channel to share your public key!
+
+### Using git-secret for Continuous Integration / Continuous Deployment (CI/CD)
+
+When using git-secret for CI/CD, you get the benefit that any deployment is necessarily done with the correct configuration, since it is collocated
+with the changes in your code.
+
+One way of doing it is the following :
+
+1. [create a gpg key](#using-gpg) for your CI/CD environment. You can chose any name and email address you want: for instance `MyApp CodeShip <myapp@codeship.com>`
+if your app is called MyApp and your CI/CD provider is CodeShip. It is easier not to define a password for that key.
+2. run `gpg --export-secret-key myapp@codeship.com --armor` to get your private key value
+3. Create an env var on your CI/CD server `GPG_PRIVATE_KEY` and assign it the private key value.
+4. Then write your Continuous Deployement build script. For instance :
```

Comment:
"Deployment"

Reply:
Please mention what OS / distro this example is targeted at.

Also, elsewhere in the docs (see: man/man7/git-secret.7.ronn) we specifically say a password for the gpg key is "always recommended", but here it says it's "easier not to". 

Maybe mention the tradeoffs involved between using/not using a password, and/or how a user would (appropriately) use a password with their keypair? Just so we're not recommending something in one place and discouraging it in another.

Reply:
I know this a touchy subject ...

What I was thinking was that if you're going to store the private key in a given env var and the password in an other env var, but store both env vars on your same CI/CD server, you do not have any additionnal security! any user having access to one env var necessarily has access to the other one too. 

At first I did set a password, and it proved to be not that easy to provide the password to all command lines that required it (I did not look for long) since we are in a no-prompt environment.

So ultimately I decided that there was no less security not having a password and that it was "easier".

I'm no security expert, so I didn't feel comfortable writing all this in a public documentation, but if you think my reasoning is good, I'll add it.

Reply:
Either change it to something simple explaining what you mean a little more, or leave it alone and we'll merge it.

Reply:
I'd like to merge this, can you fix the "Deployement" typo

Reply:
Sure! I'll do this quickly.

Reply:
Here you are 👍